### DISCUSSION TITLE
USBMSC "multiple definition" errors for ESP32 S2 MINI

### DISCUSSION THREAD

**Discussion by Zackert78** [adafruit/Adafruit_TinyUSB_Arduino#272] [Q&A]
Hey all,

Tried to upload the "midi_test" sketch to my ESP32 S2 MINI custom board and getting the following error:

`c:/users/zekie/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s2-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32s2-elf/8.4.0/../../../../xtensa-esp32s2-elf/bin/ld.exe: C:\Users\Zekie\AppData\Local\Temp\arduino\cores\00b795febbd65cae9e9635883038114b\core.a(USBMSC.cpp.o): in function tud_msc_inquiry_cb:
C:\Users\Zekie\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\cores\esp32/USBMSC.cpp:71: multiple definition of tud_msc_inquiry_cb; C:\Users\Zekie\AppData\Local\Temp\arduino\sketches\1B8248B42D1CFE423FD0FFEF3990F984\libraries\Adafruit_TinyUSB_Library\arduino\msc\Adafruit_USBD_MSC.cpp.o:c:\Users\Zekie\Documents\Arduino\libraries\Adafruit_TinyUSB_Library\src\arduino\msc/Adafruit_USBD_MSC.cpp:149: first defined here
c:/users/zekie/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s2-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32s2-elf/8.4.0/../../../../xtensa-esp32s2-elf/bin/ld.exe: C:\Users\Zekie\AppData\Local\Temp\arduino\cores\00b795febbd65cae9e9635883038114b\core.a(USBMSC.cpp.o): in function tud_msc_test_unit_ready_cb:
C:\Users\Zekie\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\cores\esp32/USBMSC.cpp:81: multiple definition of tud_msc_test_unit_ready_cb; C:\Users\Zekie\AppData\Local\Temp\arduino\sketches\1B8248B42D1CFE423FD0FFEF3990F984\libraries\Adafruit_TinyUSB_Library\arduino\msc\Adafruit_USBD_MSC.cpp.o:c:\Users\Zekie\Documents\Arduino\libraries\Adafruit_TinyUSB_Library\src\arduino\msc/Adafruit_USBD_MSC.cpp:172: first defined here
c:/users/zekie/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s2-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32s2-elf/8.4.0/../../../../xtensa-esp32s2-elf/bin/ld.exe: C:\Users\Zekie\AppData\Local\Temp\arduino\cores\00b795febbd65cae9e9635883038114b\core.a(USBMSC.cpp.o): in function tud_msc_capacity_cb:
C:\Users\Zekie\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\cores\esp32/USBMSC.cpp:89: multiple definition of tud_msc_capacity_cb; C:\Users\Zekie\AppData\Local\Temp\arduino\sketches\1B8248B42D1CFE423FD0FFEF3990F984\libraries\Adafruit_TinyUSB_Library\arduino\msc\Adafruit_USBD_MSC.cpp.o:c:\Users\Zekie\Documents\Arduino\libraries\Adafruit_TinyUSB_Library\src\arduino\msc/Adafruit_USBD_MSC.cpp:193: first defined here
c:/users/zekie/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s2-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32s2-elf/8.4.0/../../../../xtensa-esp32s2-elf/bin/ld.exe: C:\Users\Zekie\AppData\Local\Temp\arduino\cores\00b795febbd65cae9e9635883038114b\core.a(USBMSC.cpp.o): in function tud_msc_read10_cb:
C:\Users\Zekie\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\cores\esp32/USBMSC.cpp:116: multiple definition of tud_msc_read10_cb; C:\Users\Zekie\AppData\Local\Temp\arduino\sketches\1B8248B42D1CFE423FD0FFEF3990F984\libraries\Adafruit_TinyUSB_Library\arduino\msc\Adafruit_USBD_MSC.cpp.o:c:\Users\Zekie\Documents\Arduino\libraries\Adafruit_TinyUSB_Library\src\arduino\msc/Adafruit_USBD_MSC.cpp:249: first defined here
c:/users/zekie/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s2-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32s2-elf/8.4.0/../../../../xtensa-esp32s2-elf/bin/ld.exe: C:\Users\Zekie\AppData\Local\Temp\arduino\cores\00b795febbd65cae9e9635883038114b\core.a(USBMSC.cpp.o): in function tud_msc_write10_cb:
C:\Users\Zekie\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\cores\esp32/USBMSC.cpp:130: multiple definition of tud_msc_write10_cb; C:\Users\Zekie\AppData\Local\Temp\arduino\sketches\1B8248B42D1CFE423FD0FFEF3990F984\libraries\Adafruit_TinyUSB_Library\arduino\msc\Adafruit_USBD_MSC.cpp.o:c:\Users\Zekie\Documents\Arduino\libraries\Adafruit_TinyUSB_Library\src\arduino\msc/Adafruit_USBD_MSC.cpp:262: first defined here
c:/users/zekie/appdata/local/arduino15/packages/esp32/tools/xtensa-esp32s2-elf-gcc/esp-2021r2-patch5-8.4.0/bin/../lib/gcc/xtensa-esp32s2-elf/8.4.0/../../../../xtensa-esp32s2-elf/bin/ld.exe: C:\Users\Zekie\AppData\Local\Temp\arduino\cores\00b795febbd65cae9e9635883038114b\core.a(USBMSC.cpp.o): in function tud_msc_scsi_cb:
C:\Users\Zekie\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.7\cores\esp32/USBMSC.cpp:145: multiple definition of tud_msc_scsi_cb; C:\Users\Zekie\AppData\Local\Temp\arduino\sketches\1B8248B42D1CFE423FD0FFEF3990F984\libraries\Adafruit_TinyUSB_Library\arduino\msc\Adafruit_USBD_MSC.cpp.o:c:\Users\Zekie\Documents\Arduino\libraries\Adafruit_TinyUSB_Library\src\arduino\msc/Adafruit_USBD_MSC.cpp:206: first defined here
collect2.exe: error: ld returned 1 exit status
exit status 1
Compilation error: exit status 1`

I'm fairly new to the ecosystem so any help is appreciated. Thanks!

**Comment by hathach** [comment]
duplicated to #284 and should be fixed with https://github.com/adafruit/Adafruit_TinyUSB_Arduino/releases/tag/2.1.0
